configure.ac

```diff
@@ -135,6 +135,9 @@ AC_ARG_WITH([systemdsystemunitdir],
 AC_ARG_ENABLE(debug,
               [  --enable-debug      Enable debugging code.],, enable_debug="no")
 
+AC_ARG_ENABLE(extra-warnings,
+              [  --disable-extra-warnings      Disable extra compiler warnings.],, enable_extra_warnings="yes")
+
 AC_ARG_ENABLE(env-wrapper,
```

Comment:
We agreed with @dnsjts that we should default this to no, I will send a fixup soon. I want to add dozens of further warnings under this condition. Are you absolutely positive that we should separate `-Wall` from the rest? I would list all of them here (or optionally pass each of them from the configure command line, but that's a bit overkill for our purposes).